### ISSUE TITLE
sizeof(oVel) returns the size of a pointer

### ISSUE THREAD

**Issue by ghost** [jwatte/EPIC#3]
https://github.com/jwatte/EPIC/blob/master/Extrapolator.cpp#L126

`oVel` is an argument and passed as pointer to `ReadPosition()`,  so `sizeof(oVel)` returns the size of a pointer.

**Fix**

Replace `sizeof(oVel)` with `sizeof(Type) * Count`

~~~cpp
memset(oVel, 0, sizeof(Type) * Count);
~~~

**Comment by jwatte** [comment]
#include <stdio.h>
  
  template<int Count, typename Type>
  struct bar {
      void func(Type x[Count]);
  };
  
  template<int Count, typename Type> void bar<Count, Type>::func(Type x[Count]) {
    printf("%d\n", (int)sizeof(x));
  }
  
  int main() {
      int x[10];
      bar<10, int> b;
      b.func(x);
      return 0;
  }

`foo.cpp:9:29: warning: ‘sizeof’ on array function parameter ‘x’ will return size of ‘int*’`

Never a dull moment in C++ world.

Thanks for the report!

**Comment by ghost** [comment]
Yes, GCC and Clang do an amazing job. I just noticed it, because I compiled a program with Clang. VS2017 didn't complain about that (didn't try with VS2019 yet).

Thanks
